process_report/invoices/billable_invoice.py

```diff
@@ -0,0 +1,186 @@
+            pi_age = self._get_pi_age(old_pi_df, pi, invoice_month)
+            pi_old_pi_entry = old_pi_df.loc[
+                old_pi_df[invoice.PI_PI_FIELD] == pi
+            ].squeeze()
+
+            if pi_age > 1:
+                for i, row in pi_projects.iterrows():
+                    data.at[i, invoice.BALANCE_FIELD] = row[invoice.COST_FIELD]
+            else:
+                if pi_age == 0:
+                    if len(pi_old_pi_entry) == 0:
+                        pi_entry = [pi, invoice_month, new_pi_credit_amount, 0, 0]
+                        old_pi_df = pandas.concat(
+                            [
+                                pandas.DataFrame([pi_entry], columns=old_pi_df.columns),
+                                old_pi_df,
+                            ],
+                            ignore_index=True,
+                        )
+                        pi_old_pi_entry = old_pi_df.loc[
+                            old_pi_df[invoice.PI_PI_FIELD] == pi
+                        ].squeeze()
+
+                    remaining_credit = new_pi_credit_amount
+                    credit_used_field = invoice.PI_1ST_USED
+                elif pi_age == 1:
+                    remaining_credit = (
+                        pi_old_pi_entry[invoice.PI_INITIAL_CREDITS]
+                        - pi_old_pi_entry[invoice.PI_1ST_USED]
+                    )
+                    credit_used_field = invoice.PI_2ND_USED
+
+                initial_credit = remaining_credit
+                for i, row in pi_projects.iterrows():
+                    if (
+                        remaining_credit == 0
+                        or row[invoice.SU_TYPE_FIELD] in EXCLUDE_SU_TYPES
+                    ):
+                        data.at[i, invoice.BALANCE_FIELD] = row[invoice.COST_FIELD]
+                    else:
+                        project_cost = row[invoice.COST_FIELD]
+                        applied_credit = min(project_cost, remaining_credit)
+
+                        data.at[i, invoice.CREDIT_FIELD] = applied_credit
+                        data.at[i, invoice.CREDIT_CODE_FIELD] = new_pi_credit_code
+                        data.at[i, invoice.BALANCE_FIELD] = (
+                            row[invoice.COST_FIELD] - applied_credit
+                        )
+                        remaining_credit -= applied_credit
+
+                credits_used = initial_credit - remaining_credit
+                if (pi_old_pi_entry[credit_used_field] != 0) and (
+                    credits_used != pi_old_pi_entry[credit_used_field]
+                ):
+                    print(
+                        f"Warning: PI file overwritten. PI {pi} previously used ${pi_old_pi_entry[credit_used_field]} of New PI credits, now uses ${credits_used}"
+                    )
+                old_pi_df.loc[
+                    old_pi_df[invoice.PI_PI_FIELD] == pi, credit_used_field
+                ] = credits_used
+
+        old_pi_df = old_pi_df.astype(
```

Comment:
Move this to `_prepare_export`.

Reply:
@QuanMPhm if you moved this please resolve conversation.